pkg/agent/dispatcher.go

```diff
@@ -233,20 +248,47 @@ func (disp *Dispatcher) HandleMessage(
 		return fmt.Errorf("error extracting 'id field: %w", err)
 	}
 	id := uint64(*f)
+
+	var rootErr error
+
 	// now that we have the waiter's ID, make sure that if there's some failure past this point, we
 	// propagate that along to the monitor and remove it
 	defer func() {
+		// speculatively determine the root error, to send that along to the instance of Call
+		// waiting for it.
+		var err error
+
+		panicPayload := recover()
+		if panicPayload != nil {
+			err = errors.New("panicked")
+		} else if rootErr != nil {
+			err = rootErr
+		} else {
+			err = errors.New("unknown")
```

Comment:
Wouldn't we want to just return here as we're neither panicking nor is there an error?

Reply:
Hmm ok, I would just add a comment then. Or maybe even set the error message to `unknown error, did you forget to set rootError?`. Might be a bit too cheeky lol

Reply:
added a comment in 928d423.